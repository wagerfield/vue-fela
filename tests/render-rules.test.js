import { createRenderer } from 'fela'
import { renderRules } from '../index'
import rules from './component-rules'

const fela = createRenderer()
const props = {
  outerRadius: 0,
  innerRadius: 0,
  active: false
}

const testSnapshot = (optMap) => {
  expect(renderRules(fela, rules, props, optMap)).toMatchSnapshot()
}

describe('renderRules(renderer, rules, props, optMap)', () => {

  beforeEach(fela.clear)

  it('returns an object', () => {
    expect(renderRules(fela, rules, props)).toEqual(expect.any(Object))
  })

  it('renders expected snapshot', () => {
    testSnapshot()
  })

  it('renders expected snapshot when optMap is an array', () => {
    testSnapshot([ 'border' ])
  })

  it('renders expected snapshot when optMap is an object', () => {
    testSnapshot({ borderAlias: 'border' })
  })
})
