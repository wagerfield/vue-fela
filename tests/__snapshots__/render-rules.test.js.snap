// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderRules(renderer, rules, props, optMap) renders expected snapshot 1`] = `
Object {
  "border": "a b c",
  "layout": "d e f",
}
`;

exports[`renderRules(renderer, rules, props, optMap) renders expected snapshot when optMap is an array 1`] = `
Object {
  "border": "a b c",
}
`;

exports[`renderRules(renderer, rules, props, optMap) renders expected snapshot when optMap is an object 1`] = `
Object {
  "borderAlias": "a b c",
}
`;
